docs/tutorial_fixtures.rst

```diff
@@ -0,0 +1,275 @@
+
+.. literalinclude:: ../tutorials/fixtures/osu_benchmarks.py
+   :lines: 11-22
+   :emphasize-lines: 8
+
+There is nothing special to this test -- it is just an ordinary test -- except that we force it to execute locally by setting its :attr:`~reframe.core.pipeline.RegressionTest.local` variable.
+The reason for that is that a fixture at session scope can execute with any partition/environment combination, so ReFrame could have to spawn a job in case it has chosen a remote partition to launch this fixture on.
+For this reason, we simply force it execute locally regardless of the chosen partition.
+
+It is now time to run the new tests, but let us first list them:
+
+.. code-block:: bash
+
+   export RFM_CONFIG_FILE=$(pwd)/tutorials/config/settings.py
+   reframe -c tutorials/fixtures/osu_benchmarks.py -l
+
+.. code-block:: console
+
+   [ReFrame Setup]
+     version:           3.9.0
+     command:           'reframe -c tutorials/fixtures/osu_benchmarks.py -l'
+     launched by:       user@daint106
+     working directory: '/users/user/Devel/reframe'
+     settings file:     '/users/user/Devel/reframe/tutorials/config/settings.py'
+     check search path: '/users/user/Devel/reframe/tutorials/fixtures/osu_benchmarks.py'
+     stage directory:   '/users/user/Devel/reframe/stage'
+     output directory:  '/users/user/Devel/reframe/output'
+
+   [List of matched checks]
+   - osu_latency_test (found in '/users/user/Devel/reframe/tutorials/fixtures/osu_benchmarks.py')
+   - osu_allreduce_test_8 (found in '/users/user/Devel/reframe/tutorials/fixtures/osu_benchmarks.py')
+   - osu_allreduce_test_2 (found in '/users/user/Devel/reframe/tutorials/fixtures/osu_benchmarks.py')
+   - osu_allreduce_test_4 (found in '/users/user/Devel/reframe/tutorials/fixtures/osu_benchmarks.py')
+   - osu_bandwidth_test (found in '/users/user/Devel/reframe/tutorials/fixtures/osu_benchmarks.py')
+   - osu_allreduce_test_16 (found in '/users/user/Devel/reframe/tutorials/fixtures/osu_benchmarks.py')
+   Found 6 check(s)
+
+   Log file(s) saved in '/tmp/rfm-dlkc1vb_.log'
+
+Notice that only the leaf tests are listed and not their fixtures.
+Listing the tests in detailed mode, however, using the :option:`-L` option, you will see all the generated fixtures:
+
+
+.. code-block:: bash
+
+   reframe -c tutorials/fixtures/osu_benchmarks.py -n osu_bandwidth_test -L
+
+ReFrame will generate 4 fixtures for this test based on the partition and environment configurations for the current system.
+The following figure shows the generated fixtures as well as their conceptual dependencies.
+
+.. figure:: _static/img/fixtures-conceptual-deps.svg
+  :align: center
+
+  :sub:`Expanded fixtures and dependencies for the OSU benchmarks example.`
+
+Notice how the :class:`build_osu_benchmarks` fixture is populated three times, for each partition and environment combination, and the :class:`fetch_osu_benchmarks` is generated only once.
```

Comment:
```suggestion
Notice how the :class:`build_osu_benchmarks` fixture is populated three times, one for each partition and environment combination, and the :class:`fetch_osu_benchmarks` is generated only once.
```
You can also leave as it was if you prefer it.

Reply:
Perhaps `once` is better.